### ISSUE TITLE
Mouse Drags to one side

### ISSUE THREAD

**Issue by t-richter** [debauchee/barrier#112]
### Operating Systems ###

Server: macOS 10.13.6

Client: Windows 10 Version 1803 (Build 17134.137)

### Barrier Version ###

2.1.0 0b2dfd80

### Steps to reproduce bug ###

1. Switch to client
2. Move mouse to center and draw little circles
3. 9 out of 10 times Mouse drifts to one side (mostly upper left corner)

### Other info ###

* When did the problem start to occur? Right from the start
* Is there a way to work around it? No
* Does this bug prevent you from using Barrier entirely? Most likely
* Relative Mouse movement option doesn't affect it
* Tried both usb and bluetooth mouse (magic mouse)
* Moving mouse left / up feels right. opposite directions feel dragging.

**Comment by walker0643** [comment]
Can anyone confirm this? My access to a mac is lacking at the moment.

**Comment by AdrianKoshka** [comment]
I have no access to macs either, can't, sorry.

**Comment by bigfatbrowncat** [comment]
I can confirm.
Server on OSX 10.13, client on Windows 10. Mouse drags to upper left corner.

**Comment by kived** [comment]
Can also confirm with OS X 10.14 (Mojave) server and Windows 10 client. Also the mouse really, **really** wants to move up/down or left/right and not diagonal.

**Comment by wkunker** [comment]
Same issue here. When my Mojave machine is a server, my Linux client has the mouse drag problem in two directions, but not in the other two directions. When my Linux machine is a server, my Mojave machine does not have this problem, but sadly I don't get the Mac mouse acceleration. Otherwise, it works great.

**Comment by andrew-motorcentral** [comment]
Looks to be the same issue as https://symless.com/forums/topic/2036-mouse-on-client-moves-2x-as-fast-up-and-left/

**Comment by dingus9** [comment]
I am also having this issue https://symless.com/forums/topic/2036-mouse-on-client-moves-2x-as-fast-up-and-left/ , and also have a steel series sensei raw mouse. However switching to a logitec RX1000 also shows similar behavior only the up and left movement creep is about half of the sensei mouse. The trackpad on the server is more usable, however it too is gradually moving up and left.
```
Server: OSX mojave 10.14.4, sensei raw mouse and Logitech RX1000
Client OSX 10.13.6
```

**Comment by AdrianKoshka** [comment]
[New builds for windows available](https://github.com/debauchee/barrier/releases/tag/v2.1.2)

**Comment by andrew-motorcentral** [comment]
Thanks @AdrianKoshka - I have just installed the new build (2.2.0-Release-18ac0fe4) and the top-left mouse creep issue still occurs (macOS Mojave 10.14.4 -> Windows 10 Pro 1809). 

I can confirm that this occurs with both an Apple Magic Mouse and an Acer standard optical mouse (though the Magic Mouse seems to creep faster)

**Comment by AdrianKoshka** [comment]
If anyone is still having this issue, we have a new macOS build, and I'm curious if it fixes it for you: https://github.com/debauchee/barrier/releases/tag/v2.3.0

**Comment by below43** [comment]
Thanks @AdrianKoshka 

I am sorry, but this issue still occurs.

**Comment by AdrianKoshka** [comment]
No problem.

**Comment by bartoncasey** [comment]
Looks like this is addressed in https://github.com/symless/synergy-core/pull/6469 .

**Comment by bartoncasey** [comment]
I just confirmed that it does fix the issue on my machine.

**Comment by AdrianKoshka** [comment]
Ugh, the branch used for the PR was deleted so it's not as easy as creating a PR.

**Comment by dingus9** [comment]
cherry pick? https://github.com/symless/synergy-core/pull/6469/commits/2bbfe450aecffdbd1283f3d568e1cc9c66dd327c

**Comment by ryoplasmic** [comment]
Hi, having the same issue on macOS 10.13.6 (server) and Windows 10 (client) on version 2.3.2 

macOS side macbook trackpad and magic mouse have no issue. Only have issue when use with Logitech G502/G502Hero. Basically, when moving the mouse in the client, it will drift faster and will suddenly jumped back to the server. It doesn't happens when using Windows 10 as server.

No matter which mouse to use, when moved the cursor to client, the server still shows the cursor in the middle of the screen. This happens in the external screen. (my set up is Macbook laptop + external screen & Windows laptop + external screen)

**Comment by below43** [comment]
@ryoplasmic That sounds like a completely different issue (which was a very slow mouse drift, not as you describe).

v2.3.2 resolved this issue (it included Merge #424)

**Comment by p12tic** [comment]
It seems that #424 has fixed this issue. Please reopen if that's not the case.
